code/ui/blocks/src/blocks/Primary.tsx

```diff
@@ -11,17 +12,58 @@ interface PrimaryProps {
    * @deprecated Primary block should only be used to render the primary story, which is automatically found.
    */
   name?: string;
+  /**
+   * Specify where to get the primary story from.
+   */
+  of?: Of;
 }
 
-export const Primary: FC<PrimaryProps> = ({ name }) => {
+const getPrimaryFromResolvedOf = (resolvedOf: ReturnType<typeof useOf>) => {
+  switch (resolvedOf.type) {
+    case 'meta': {
+      return resolvedOf.csfFile.stories[0] || null;
+    }
+    case 'component': {
+      throw new Error(
+        `Unsupported module type. Primary's \`of\` prop only supports \`meta\`, got: ${
+          (resolvedOf as any).type
+        }`
+      );
+    }
+    default: {
+      throw new Error(
+        `Unrecognized module type resolved from 'useOf', got: ${(resolvedOf as any).type}`
+      );
+    }
+  }
+};
+
+export const Primary: FC<PrimaryProps> = (props) => {
+  const { name, of } = props;
+
+  if ('of' in props && of === undefined) {
+    throw new Error('Unexpected `of={undefined}`, did you mistype a CSF file reference?');
+  }
+
   const docsContext = useContext(DocsContext);
+
+  let story;
+  if (of) {
+    const resolvedOf = useOf(of || 'meta');
```

Comment:
The second argument to `useOf` is an array of valid types, here it would make sense to specify `'meta'` as the only valid type.

```suggestion
    const resolvedOf = useOf(of || 'meta', ['meta']);
```

Reply:
Ok thanks for your review, good catch! I'll push a commit asap.